api/src/main/java/org/apache/iceberg/metrics/Counter.java

```diff
@@ -59,7 +59,7 @@ default String name() {
   }
 
   Counter NOOP =
-      new Counter() {
+      new DefaultCounter("undefined", Unit.COUNT) {
```

Comment:
nit: I've updated the unit to `Unit.UNDEFINED` as I think that's more appropriate

Reply:
ErrorProne actually complains about this 
```
/home/nastra/Development/workspace/iceberg/api/src/main/java/org/apache/iceberg/metrics/Counter.java:69: error: [ClassInitializationDeadlock] Static fields and blocks within a class must not require a subclass, otherwise a deadlock occurs when separate threads simultaneously attempt to initialize the subclass and base class. This behavior is allowed by the java language specification and is not considered a Java bug.
      new DefaultCounter("undefined", Unit.UNDEFINED) {
      ^
  From https://docs.oracle.com/javase/specs/jls/se8/html/jls-12.html#jls-12.4.1
  > 12.4.1. When Initialization Occurs
  > A class or interface type T will be initialized immediately before the first occurrence of any one of the following:
  > • T is a class and an instance of T is created.
  > • A static method declared by T is invoked.
  > • A static field declared by T is assigned.
  > • A static field declared by T is used and the field is not a constant variable (§4.12.4).
  > • T is a top level class (§7.6) and an assert statement (§14.10) lexically nested within T (§8.1.3) is executed.
    (see https://github.com/palantir/gradle-baseline#baseline-error-prone-checks)
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error
```

Reply:
this is a real bummer :( I guess we'll have to keep the IntCounter/LongCounter until we eventually remove all those things that are deprecated

Reply:
Could we get around this by using a `static Counter noop()` method that is lazily initialized?

Reply:
I think we'd have to define the noop counter outside Counter itself to avoid the init deadlock